src/v/debug_bundle/debug_bundle_service.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "base/seastarx.h"
+#include "config/property.h"
+#include "debug_bundle/error.h"
+
+#include <seastar/core/sharded.hh>
+
+namespace debug_bundle {
+
+/**
+ * @brief Service used to manage creation of debug bundles
+ *
+ * This service is used to create, delete, and manage debug bundles using the
+ * "rpk debug bundle" application
+ */
+class debug_bundle_service final
+  : public ss::peering_sharded_service<debug_bundle_service> {
+public:
+    /// Default shard operations will be performed on
+    static constexpr ss::shard_id service_shard = 0;
+    /// Name of the debug bundle directory
+    static constexpr std::string_view debug_bundle_dir_name = "debug-bundle";
```

Comment:
It might be worth making this configurable, so that it can go on a different mount than the data dir.

Reply:
I think I want to come back to this after doing some thinking on it.  I would prefer to be able to "clean up" if the data directory does change.  Let's table this for later on, if that's ok?